desktop-widgets/filterwidget2.cpp

```diff
@@ -68,23 +75,24 @@ FilterWidget2::FilterWidget2(QWidget* parent)
 
 void FilterWidget2::updateFilter()
 {
-	FilterData data {
-		ui->minVisibility->currentStars(),
-		ui->maxVisibility->currentStars(),
-		ui->minRating->currentStars(),
-		ui->maxRating->currentStars(),
-		ui->minWaterTemp->value(),
-		ui->maxWaterTemp->value(),
-		ui->minAirTemp->value(),
-		ui->maxWaterTemp->value(),
-		ui->from->dateTime(),
-		ui->to->dateTime(),
-		ui->tags->text().split(",", QString::SkipEmptyParts),
-		ui->people->text().split(",", QString::SkipEmptyParts),
-		ui->location->text().split(",", QString::SkipEmptyParts),
-		ui->equipment->text().split(",", QString::SkipEmptyParts),
-		ui->invertFilter->isChecked()
-	};
+	FilterData data;
+
+	data.validFilter = true;
+	data.minVisibility = ui->minVisibility->currentStars();
+	data.maxVisibility = ui->maxVisibility->currentStars();
+	data.minRating = ui->minRating->currentStars();
+	data.maxRating = ui->maxRating->currentStars();
+	data.minWaterTemp = ui->minWaterTemp->value();
+	data.maxWaterTemp = ui->maxWaterTemp->value();
+	data.minAirTemp = ui->minAirTemp->value();
+	data.maxWaterTemp = ui->maxWaterTemp->value();
+	data.from = ui->from->dateTime();
+	data.to = ui->to->dateTime();
+	data.tags = ui->tags->text().split(",", QString::SkipEmptyParts);
+	data.people = ui->people->text().split(",", QString::SkipEmptyParts);
+	data.location = ui->location->text().split(",", QString::SkipEmptyParts);
+	data.equipment = ui->equipment->text().split(",", QString::SkipEmptyParts);
+	data.invertFilter = ui->invertFilter->isChecked();
```

Comment:
again, this should be cleaned up. You introduce new code, then you introduce a new data structure, then you change the code you introduced 6 commits ago to use the new data structure. Let's not do that.